lager/event_loop/qapplication.hpp

```diff
@@ -0,0 +1,48 @@
+//
+// lager - library for functional interactive c++ programs
+// Copyright (C) 2019 Carl Bussey
+//
+// This file is part of lager.
+//
+// lager is free software: you can redistribute it and/or modify
+// it under the terms of the MIT License, as detailed in the LICENSE
+// file located at the root of this source code distribution,
+// or here: <https://github.com/arximboldi/lager/blob/master/LICENSE>
+//
+
+#pragma once
+
+#include <QApplication>
+#include <QMetaObject>
+#include <Qt>
+#include <QtConcurrentRun>
+
+#include <functional>
+#include <utility>
+
+namespace lager {
+
+struct with_qapplication_event_loop
+{
+    std::reference_wrapper<QApplication> app;
+
+    template <typename Fn>
+    void async(Fn&& fn)
+    {
+        QtConcurrent::run(std::forward<Fn>(fn));
```

Comment:
It would be nice to have something to garantee that async operations are finished at some point before the desutrction of the store. We did something like that for `iostreams`. We do not need to do it now though.